### PR TITLE
Update grammar/docs for current fixed point types

### LINKED ISSUE #2571: Update ABI/grammar/docs for new fixed point type encoding
- documentation (the types section is currently empty)
- ABI
- `docs/grammar.txt`
- `docs/utils/SolidityLexer.py`

Follow up of #1544.

### PR DESCRIPTION
**Pull request by axic**
Fixes #2571.

### CODE REVIEW

File: docs/grammar.txt [hunk 1]
```diff
@@ -142,9 +142,11 @@ Uint = 'uint' | 'uint8' | 'uint16' | 'uint24' | 'uint32' | 'uint40' | 'uint48' |
...
+FixedSize = ( '8' | '16' | '24' | '32' | '40' | '48' | '56' | '64' | '72' | '80' | '88' | '96' | '104' | '112' | '120' | '128' | '136' | '144' | '152' | '160' | '168' | '176' | '184' | '192' | '200' | '208' | '216' | '232' | '240' | '248' | '256' )
```
Comment (axic) [inline docs/grammar.txt]: I am not fully sure it makes sense restricting the types on the grammar level, it just results in a giant list and certain combinations are currently ruled out in type checking.

Parser only checks that `W` must be between 8 and 256 and divisible by 8, but `F` is only checked for <= 80.

We could decide to make all `W`/`F` valid are raise an error in the type checker (same applies to `int`, `uint`, `bytes`)

**Comment by federicobond** [comment]
The rules for fixed types are lexer rules (they describe a single token with no spaces or comments between its components). We should try to clarify that for parser implementers.

**Comment by axic** [comment]
To be honest I think the grammar should only have a list of builtin types in the format of `uint` + decimal, etc.

File: docs/grammar.txt (hunk 1 above)
Comment (axic) [reply docs/grammar.txt]: @chriseth can we just go for `DecimalNumber 'x' DecimalNumber`?

Comment (chriseth) [reply docs/grammar.txt]: 👍

**Comment by axic** [comment]
@chriseth done

**Review by chriseth — APPROVED** [review]

**Pull request merged**